src/NodeApi/IJSValue.cs

```diff
@@ -1,41 +1,80 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-#if !NET7_0_OR_GREATER
 using System;
+
+#if !NET7_0_OR_GREATER
 using System.Reflection;
 #endif
 
 namespace Microsoft.JavaScript.NodeApi;
 
-#if NET7_0_OR_GREATER
-// A static interface that helps with the conversion of JSValue to a specific type.
-public interface IJSValue<TSelf> where TSelf : struct, IJSValue<TSelf>
+/// <summary>
+/// A base interface for a struct that represents a JavaScript value type or a built-in
+/// object type. It provides functionality for converting between the struct
+/// and <see cref="JSValue"/>.
+/// </summary>
+/// <typeparam name="TSelf">The derived struct type.</typeparam>
+public interface IJSValue<TSelf> : IEquatable<JSValue> where TSelf : struct, IJSValue<TSelf>
 {
-    public static abstract bool CanBeConvertedFrom(JSValue value);
+    /// <summary>
+    /// Converts the derived struct `TSelf` to a <see cref="JSValue"/> with
+    /// the same `napi_value` handle.
+    /// </summary>
+    /// <returns>
+    /// <see cref="JSValue"/> with the same `napi_value` handle as the derived struct.
+    /// </returns>
+    JSValue AsJSValue();
 
-    public static abstract TSelf CreateUnchecked(JSValue value);
+#if NET7_0_OR_GREATER
+    /// <summary>
+    /// Checks id the derived struct `TSelf` can be created from a <see cref="JSValue"/>.
+    /// </summary>
+    static abstract bool CanCreateFrom(JSValue value);
+
+    /// <summary>
+    /// Creates a new instance of the derived struct `TSelf` from a <see cref="JSValue"/> without
+    /// checking the enclosed handle type.
+    /// </summary>
+    static abstract TSelf CreateUnchecked(JSValue value);
+#endif
 }
-#else
-// A static class that helps with the conversion of JSValue to a specific type.
-public static class IJSValueShim<T> where T : struct
+
+#if !NET7_0_OR_GREATER
+/// <summary>
+/// Implements IJSValue interface static functions for the previous .Net versions.
+/// </summary>
+/// <typeparam name="T"></typeparam>
+internal static class IJSValueShim<T> where T : struct, IJSValue<T>
 {
-    private static readonly Func<JSValue, bool> s_canBeConvertedFrom =
+    /// <summary>
+    /// A static field to keep a reference to the CanCreateFrom private method.
+    /// </summary>
+    private static readonly Func<JSValue, bool> s_canBeCreatedFrom =
         (Func<JSValue, bool>)Delegate.CreateDelegate(
             typeof(Func<JSValue, bool>),
             typeof(T).GetMethod(
-                nameof(JSObject.CanBeConvertedFrom),
-                BindingFlags.Static | BindingFlags.Public)!);
+                "CanCreateFrom",
```

Comment:
`nameof(CanCreateFrom)` since there is now a method of that name in the current class.

Reply:
Fixed.